### PR TITLE
[Xamarin.Android.Build.Tasks] NullReferenceException in Xamarin.Android.Common.targets

### LINKED ISSUE #6124: NullReferenceException in Xamarin.Android.Common.targets
### Steps to Reproduce

1. See for details https://github.com/taori/PCRemote2/runs/3153046999?check_suite_focus=true
2. See run file for msbuild command which forwards github secrets to msbuild command

### Expected Behavior
A signed package should be created
### Actual Behavior
Build fails because of NRE
### Log File

```
"D:\a\PCRemote2\PCRemote2\src\Amusoft.PCR.Mobile.Droid\Amusoft.PCR.Mobile.Droid.csproj" (SignAndroidPackage target) (1:7) ->
(_Sign target) -> 
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2300,2): error XAAAS7001: System.NullReferenceException: Object reference not set to an instance of an object. [D:\a\PCRemote2\PCRemote2\src\Amusoft.PCR.Mobile.Droid\Amusoft.PCR.Mobile.Droid.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2300,2): error XAAAS7001:    at Xamarin.Android.Tasks.AndroidApkSigner.AddStorePass(CommandLineBuilder cmd, String cmdLineSwitch, String value) [D:\a\PCRemote2\PCRemote2\src\Amusoft.PCR.Mobile.Droid\Amusoft.PCR.Mobile.Droid.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2300,2): error XAAAS7001:    at Xamarin.Android.Tasks.AndroidApkSigner.GenerateCommandLineCommands() [D:\a\PCRemote2\PCRemote2\src\Amusoft.PCR.Mobile.Droid\Amusoft.PCR.Mobile.Droid.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2300,2): error XAAAS7001:    at Microsoft.Build.Utilities.ToolTask.Execute() [D:\a\PCRemote2\PCRemote2\src\Amusoft.PCR.Mobile.Droid\Amusoft.PCR.Mobile.Droid.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2300,2): error XAAAS7001:    at Microsoft.Android.Build.Tasks.AndroidToolTask.RunTask() in /Users/builder/azdo/_work/2/s/xamarin-android/external/xamarin-android-tools/src/Microsoft.Android.Build.BaseTasks/AndroidToolTask.cs:line 24 [D:\a\PCRemote2\PCRemote2\src\Amusoft.PCR.Mobile.Droid\Amusoft.PCR.Mobile.Droid.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(2300,2): error XAAAS7001:    at Microsoft.Android.Build.Tasks.AndroidToolTask.Execute() in /Users/builder/azdo/_work/2/s/xamarin-android/external/xamarin-android-tools/src/Microsoft.Android.Build.BaseTasks/AndroidToolTask.cs:line 15 [D:\a\PCRemote2\PCRemote2\src\Amusoft.PCR.Mobile.Droid\Amusoft.PCR.Mobile.Droid.csproj]
```

### PR DESCRIPTION
**Pull request by dellis1972**
Fixes #6124

Commit 2a04748d removed the `[Required]` attribute from the `KeyPass`
and `StorePass` properties. This was because they were now optional.
However `keytool` does REQUIRE these parameters when we use the `-ks`
or `KeyStore` argument.

If a user does not pass these arguments they get the following error

```
    error XAAAS7001: System.NullReferenceException: Object reference not set to an instance of an object.
    error XAAAS7001: at Xamarin.Android.Tasks.AndroidApkSigner.AddStorePass(CommandLineBuilder cmd, String cmdLineSwitch, String value)
    error XAAAS7001: at Xamarin.Android.Tasks.AndroidApkSigner.GenerateCommandLineCommands()
```

This is because the `KeyPass` or `StorePass` are `null` and the `AddStorePass`
function is not handling that.

So what we should do is when the `KeyStore` argument is being used we should
error out if `KeyPass` or `StorePass` are empty or `null`.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Tasks/AndroidApkSigner.cs [hunk 1]
```diff
@@ -138,6 +138,14 @@ protected override bool ValidateParameters ()
...
 					return false;
 				}
+				if (string.IsNullOrEmpty (KeyPass)) {
+					Log.LogCodedError ("XA4314", Properties.Resources.XA4310, "$(AndroidSigningKeyPass)");
+					return false;
+				}
+				if (string.IsNullOrEmpty (StorePass)) {
+					Log.LogCodedError ("XA4314", Properties.Resources.XA4310, "$(AndroidSigningStorePass)");
+					return false;
+				}
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Tasks/AndroidApkSigner.cs:148]: I think these have the wrong message: `Properties.Resources.XA4310`, and the code is `XA4314`.

Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Tasks/AndroidApkSigner.cs:148]: 🤦

Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Tasks/AndroidApkSigner.cs:148]: Actually , no. I added a new code XA4314 for this specific error. The XA4310 is for files existing, in this case Xa4314 is for properties being empty or null.

Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Tasks/AndroidApkSigner.cs:148]: Maybe just add a comment, if we see this later. Someone might try to change it to make the numbers match.

Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Tasks/AndroidApkSigner.cs]: I just realised what you were talking about lol. I updated the `Properties.Resources.XA4310` to be `Properties.Resources.XA4314` as it should have been. It was a place holder until I compiled the new `resx` lol.

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**